crates/matrix-sdk-ui/src/notification_client.rs

```diff
@@ -59,6 +62,9 @@ pub struct NotificationClient {
     /// Should we try to filter out the notification event according to the push
     /// rules?
     filter_by_push_rules: bool,
+
+    /// A mutex to serialize requests to sliding sync.
```

Comment:
Why a mutex is needed?

Reply:
Mostly as a measure of caution. Imagine you're using ElementX Android, you were in airplane mode and just left it. The push gateway may send you N notifications at the same time, so the `get_notification` method may be called N times in parallel. The SS proxy requires that, given a connection id, we have *at most* one long-polling request to the server. So to handle N notifications, we have one of two choices:

- serialize the requests (which I've done here, using a mutex)
- use different connection IDs
- ??? (other methods I haven't thought of)

Reply:
Oh I know why you need a `Mutex` but the doc doesn't explain that :-p. Sorry fort the confusion. Can you improve the documentation a little bit please ?

Reply:
Ah nice, I did it before you submitted this comment :grin: